Composer/packages/ui-plugins/virtual-assistant-creation/src/components/provisionSummaryPage.tsx

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import formatMessage from 'format-message';
+import React, { Fragment } from 'react';
+import { RouteComponentProps } from '@reach/router';
+import { DialogWrapper, DialogTypes } from '@bfc/ui-shared';
+
+import { Pivot, PivotItem } from 'office-ui-fabric-react/lib/Pivot';
+import { DialogFooterWrapper } from './dialogFooterWrapper';
+import { RouterPaths } from '../shared/constants';
+import { DetailsList, DetailsListLayoutMode, IColumn, SelectionMode } from 'office-ui-fabric-react/lib/DetailsList';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+
+// -------------------- BotTypeTile -------------------- //
+interface ProvisionSummaryPageProps extends RouteComponentProps {
```

Comment:
prefer type over interface

Reply:
Can a type extend an interface? Or do you need to define an intersection type?

Reply:
RouteComponentProps itself is a type that accepts a T for props: RouteComponentProps<TProps>

Reply:
I have converted all props interfaces to types using the method @hatpick outlined. VirtualAssistantCreationModalProps example:
```
type VirtualAssistantCreationModalProps = RouteComponentProps<{
  onDismiss: () => void;
  handleCreateNew: (formData: formData, templateId: string) => Promise<void>;
  formData: formData;
}>;
```

However `RouteComponentProps` is set up in such a way that the partial props values passed to it have a type of `DECLAREDTYPE | undefined`. 

Which resulted in having to add the following undefined checking code at the start of each functional component: 
```
 const { onDismiss, handleCreateNew, formData } = props;
  if (onDismiss === undefined || handleCreateNew === undefined || formData === undefined) {
    console.log('invalid props passed to VirtualAssistantCreationModal');
    return null;
  }
```
This does not happen with the previous interface definition. So converting these RouteComponentProps interfaces to types does add a bit of additional null checking code. This has nothing to do with the type conversion but more so for types that "extend" RouteComponetProps. 

- Should I keep this as types regardless (change is currently in latest commit in this branch)?
- Alternatively I can go back to using interfaces that extend `RouteComponetProps`?
- Or get rid of reach/router in this workspace all together and handle navigation state in appState, this would remove the need to have props extend `RouteComponetProps` and may be something we want to do regardless of this issue.

Reply:
@pavolum that's a weird thing they do, I suggest keeping it as type (in oppose to interface) but do an intersection type:
```
type MyProps = {
  onDismiss: () => void;
  handleCreateNew: (formData: formData, templateId: string) => Promise<void>;
  formData: formData;
} & ReactComponentProps<{}>;
```

Reply:
Great, thanks for the tip. I have implemented this across the repo with the latest iteration.